src/__tests__/test-utils-selectors.test.tsx

```diff
@@ -0,0 +1,141 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import fs from 'fs';
+import Path from 'path';
+import glob from 'glob';
+import { transformSync, types, PluginObj, NodePath } from '@babel/core';
+import { flatten, zip } from 'lodash';
+
+// The test extracts generated selectors from the compiled output and matches those against the snapshot.
+test('test-utils selectors', () => {
+  // Find referenced selector files and properties.
+  const selectorsFilePathToUsedProperties = new Map<string, Set<string>>();
+  for (const file of glob.sync('lib/components/test-utils/selectors/**/*.js')) {
```

Comment:
Are you sure it is a good idea to exctract this info from the compiled code? Would it be easier to extract it from typescript sources? (`import` syntax is much easier to statically analyse, I can see how the code below simplifies)

Reply:
Thanks, I will try this out.